dmscripts/mark_definite_framework_results.py

```diff
@@ -108,42 +121,62 @@ def mark_definite_framework_results(
             reassess_failed_draft_services=reassess_failed_draft_services,
             logger=logger
         )
+        if not service_counter["passed"]:
+            fail_supplier(supplier_id, framework_slug, updated_by, supplier_framework, client, logger, dry_run=dry_run)
+            continue
 
-        if (
-            supplier_framework["declaration"].get("status") == "complete"
-            and service_counter["passed"]
-            and _passes_validation(
+        # Check for a definite pass
+        supplier_passes_validation = _passes_validation(
+            supplier_framework["declaration"],
+            declaration_definite_pass_schema,
+            logger,
+            schema_name="declaration_definite_pass_schema",
+            tablevel=1
+        )
+        if supplier_passes_validation:
+            # Congratulations supplier, you pass!
+            pass_supplier(supplier_id, framework_slug, updated_by, supplier_framework, client, logger, dry_run=dry_run)
+            continue
+
+        # The supplier didn't pass, but check for, and default to, a discretionary pass
+        if declaration_discretionary_pass_schema:
+            supplier_passes_discretionary_validation = _passes_validation(
                 supplier_framework["declaration"],
-                declaration_definite_pass_schema,
+                declaration_discretionary_pass_schema,
```

Comment:
Is this a change in the logic of the script? are we no longer doing definite passes?